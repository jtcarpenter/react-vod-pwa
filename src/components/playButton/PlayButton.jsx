import React from 'react';
import * as playerStates from 'constants/playerStates';

export function PlayButton({videoState, handlePlay, handlePause}) {

    const className = videoState === playerStates.PLAYING
        ? 'icon-pause player-ctrl player-ctrl__play'
        : 'icon-play player-ctrl player-ctrl__play';

    const handleClick = () => {
        if (videoState === playerStates.PLAYING) {
            return handlePause();
        }
        return handlePlay();
    }

    return (
        <button
            type="button"
            className={className}
            onClick={handleClick}
        >
        </button>
    );
}

export default PlayButton;