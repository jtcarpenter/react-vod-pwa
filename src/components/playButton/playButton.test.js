import React from 'react';
import {shallow} from 'enzyme';
import {PlayButton} from 'components/playButton/PlayButton';

const handlePlay = () => {};
const handlePause = () => {};
const player = (
    <PlayButton
        handlePlay={handlePlay}
        handlePause={handlePause}
    />
)

describe('PlayButton', () => {
    it('should render', () => {
        const wrapper = shallow(player);
        expect(wrapper.exists()).toBe(true);
    })
})