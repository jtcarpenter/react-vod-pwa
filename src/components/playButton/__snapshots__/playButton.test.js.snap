// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayButton should display correctly 1`] = `
<button
  className="sc-bdVaJa cYyHzu"
  icon="\\\\ea15"
  onClick={[Function]}
  type="button"
/>
`;
